PSI_Tracker.js

```diff
@@ -0,0 +1,645 @@
+          sheet.getRange(batch[i].id+1,COLUMN_STATUS+1).setNote("Error " + e.toString() + "\n\nFull error:\n" + JSON.stringify(data));
+          debug += "🟥 " + "Error " + e.toString() + "\n\nFull error:\n" + JSON.stringify(data) + " | " 
+          fields_for_debug.push(debug)
+          DEBUG_SHEET.appendRow(fields_for_debug);
+        }
+        Logger.log(content)
+        // If an error occured, flagged it in the Config tab, under the Status
+        if(content["error"]){
+            sheet.getRange(batch[i].id+1,COLUMN_STATUS+1).setValue("❌")
+            sheet.getRange(batch[i].id+1,COLUMN_STATUS+1).setNote("Error " + JSON.stringify(content["error"]))
+            debug += "❌ " + "Error " + JSON.stringify(content["error"]) + " | " 
+            fields_for_debug.push(debug)
+            DEBUG_SHEET.appendRow(fields_for_debug);
+        }
+        else{
+            // Build a dic that will contain all the extracted information
+            let output = {
+              "Date": new Date().toISOString().slice(0, 10),
+              "Label": batch[i].label,
+              "URL": batch[i].url,
+              "Device": batch[i].device,
+              "URL / Origin": batch[i].origin
+            };
+            // Iterate through all the fields and eval the function to get their respective values, adding them to the dic
+            for (let j = 0; j < fields.length; j++) {
+                try{
+                    // Origin field is used in some calls to know if there was no URL-level data and we reverted to Origin level
+                    let origin = batch[i].origin;
+                    let green_domains_ = green_domains;
+                    // Eval the formula, for example "content["lighthouseResult"]["lighthouseVersion"]"
+                    let value = eval(fields[j].data)
+                    // Add it to our dic
+                    output[fields[j].field] = value
+                }
+                catch(e){
+                    // If error, write it in placce -> we ultimately decided to leave it blank instead as it was breaking dashboarding down the line
+                    // output[fields[j].field] = "Error " + JSON.stringify(e.toString())
+                    output[fields[j].field] = ""
+                    debug += "❗️ Error on " + fields[j].field + JSON.stringify(e.toString()) + " | " 
+                }
+            }
+            Logger.log(output);
+            
+            // STEP 3. We build the new row based on what we read on the Results first line
+            if(api == "PSI"){
+              let sheet_results = RESULT_SHEET;
+              let results_header = RESULT_SHEET_HEADER;
+              // We translate the dic output to an array row based on indexes found in "Results"
+              let array = [];
+              // Go through all the Dimensions listed in the header and build the array
+              for (let k = 0; k < results_header[0].length; k++){
```

Comment:
It seems like we're now three levels deep in nested for loops, but I think it's only two. I'd suggest switching this to `j` if it is only two, or change `i`, `j`, and `k` to more descriptive names.

Reply:
fixed to `j` and will do a second round of review to have generic iteration with indices